ement-room.el

```diff
@@ -2817,151 +2809,113 @@ MSG is the message STRING was contained in."
-                                       'help-echo (or url "no link")
-                                       'action (lambda (d)
-                                                 (when (stringp d)
-                                                   (browse-url d)))
-                                       'follow-link (when url t)
-                                       'button t
-                                       'button-data url
-                                       'keymap button-map
-                                       'category t)
-                           "\n")))
-    (when-let ((type (alist-get 'og:type data))
-               (typestr (cond ((string-match-p "music.*" type) "Music")
-                              ((string-match-p "video.*" type) "Video")
-                              ((string-match-p "article.*" type) "Article")
-                              ((string-match-p "book.*" type) "Book")
-                              ((string-match-p "profile.*" type) "Profile"))))
-      (setf string (concat string
-                           (propertize typestr 'face 'ement-room-preview)
-                           "\n")))
-    (when-let ((desc (alist-get 'og:description data)))
-      (let* ((current-left-margin (or (car (window-margins)) 0))
-             (current-right-margin (or (cdr (window-margins)) 0))
-             (available-width (abs (- (- (window-width)
-                                         (+ current-left-margin current-right-margin))
-                                      (length ement-room-url-preview-prefix)))))
-        (setf string (concat string
-                             (propertize (truncate-string-to-width (replace-regexp-in-string "\n"
-                                                                                             " "
-                                                                                             desc)
-                                                                   available-width)
-                                         'face 'ement-room-preview)
-                             "\n"))))
-    (propertize string 'line-prefix ement-room-url-preview-prefix)))
+  (with-temp-buffer
+    (when-let ((title (alist-get 'og:title data)))
+      (insert (propertize title
+                          'mouse-face 'highlight
+                          'face '(link (:inherit ement-room-preview))
+                          'help-echo (alist-get 'og:url data)
+                          'action (lambda (d)
+                                    (when (stringp d)
+                                      (browse-url d)))
+                          'follow-link (alist-get 'og:url data)
+                          'button t
+                          'button-data (alist-get 'og:url data)
+                          'keymap button-map
+                          'category t)
+              "\n"))
+    (let* ((type? (alist-get 'og:type data))
```

Comment:
I don't mind the Scheme-style `foo?` naming in general, but Ement is an Emacs package, and I'm trying to follow Emacs's conventions, so please use, e.g. `foo-p` instead.